### DISCUSSION TITLE
Best way to do API versioning

### DISCUSSION THREAD

**Discussion by EmilioMirasola** [oapi-codegen/oapi-codegen#1372] [Q&A]
Hi,

We can't wrap our head around which way is the best to do API versioning with oapi-codegen.
Does anybody have experience with this? If yes I'd love to hear how you've done it and which pain points you've encountered😄

**Comment by jamietanna** [comment]
I've done it before - on a codebase I no longer have access to - using content-type negotiation.

More details on my reasoning behind it in https://www.jvt.me/posts/2021/01/05/why-content-negotiation/

Alternatively, you could do it with path-based versioning, and have your `v1` API in one set of files and `v2` in another, where they're `$ref`'d together.

(Very happy to talk more, just thought I'd drop this reply in here while I see it!)
